source/Calamari.Tests/KubernetesFixtures/ResourceStatus/ResourceStatusReportWrapperTests.cs

```diff
@@ -0,0 +1,235 @@
+                 {
+                     new ResourceIdentifier("ConfigMap", configMapName, "default")
+                 });
+             }
+             finally
+             {
+                 fileSystem.DeleteDirectory(tempDirectory);
+             }
+         }
+
+        [Test]
+         public void FindsSecretsDeployedInADeployContainerStep()
+         {
+             var variables = new CalamariVariables();
+             var log = new SilentLog();
+             var fileSystem = new TestCalamariPhysicalFileSystem();
+             var statusChecker = new MockResourceStatusChecker();
+        
+             const string secret = "Secret-Deployment-01";
+             AddKubernetesStatusCheckVariables(variables);
+             variables.Set("Octopus.Action.KubernetesContainers.KubernetesSecretEnabled", "True");
+             variables.Set("Octopus.Action.KubernetesContainers.ComputedSecretName", secret);
+        
+             var tempDirectory = fileSystem.CreateTemporaryDirectory();
+             try
+             {
+                 fileSystem.SetFileBasePath(tempDirectory);
+        
+                 var wrapper = new ResourceStatusReportWrapper(variables, log, fileSystem, statusChecker);
+                 wrapper.NextWrapper = new StubScriptWrapper();
+        
+                 wrapper.ExecuteScript(
+                     new Script("stub"),
+                     Syntax,
+                     new CommandLineRunner(log, variables),
+                     new Dictionary<string, string>());
+        
+                 statusChecker.CheckedResources.Should().BeEquivalentTo(new[]
+                 {
+                     new ResourceIdentifier("Secret", secret, "default")
+                 });
+             }
+             finally
+             {
+                 fileSystem.DeleteDirectory(tempDirectory);
+             }
+         }
+
+         private static void AddKubernetesStatusCheckVariables(IVariables variables)
+         {
+             variables.Set(Deployment.SpecialVariables.EnabledFeatureToggles, "KubernetesDeploymentStatusFeatureToggle");
+             variables.Set(SpecialVariables.ClusterUrl, "https://localhost");
+             variables.Set(SpecialVariables.ResourceStatusCheck, "True");
+         }
+    }
+
+    internal class MockResourceStatusChecker : IResourceStatusChecker
+    {
+        public List<ResourceIdentifier> CheckedResources { get; private set; }
+
+        public bool CheckStatusUntilCompletionOrTimeout(
+            IEnumerable<ResourceIdentifier> resourceIdentifiers, 
+            IStabilizingTimer stabilizingTimer,
+            Kubectl kubectl)
+        {
+            CheckedResources = resourceIdentifiers.ToList();
+            return true;
+        }
+    }
+
+    internal class StubScriptWrapper : IScriptWrapper
+    {
+        public int Priority { get; } = 1;
+        public IScriptWrapper NextWrapper { get; set; }
+        public bool IsEnabled(ScriptSyntax syntax) => true;
```

Comment:
Set this to be `false` by default, and then override it via a `SetEnable` method ?